packages/cli/src/commands/env/pull.ts

```diff
@@ -161,6 +165,28 @@ export default async function pull(
     // https://github.com/vercel/next.js/blob/06abd634899095b6cc28e6e8315b1e8b9c8df939/packages/create-next-app/templates/app/js/gitignore#L28
     const rootPath = link.repoRoot ?? cwd;
     isGitIgnoreUpdated = await addToGitIgnore(rootPath, '.env*.local');
+  } else {
+    const included = gitIgnoreIncludesFile(filename, gitignorePath);
+
+    if (!included) {
+      output.warn(
+        `You are pulling your Environment Variables into ${chalk.bold(filename)} which is ` +
+          `${chalk.red(chalk.bold('NOT'))} in your .gitignore. This means that any ` +
+          `sensitive values, API keys, or secrets could be commited to git.`
+      );
+
+      if (!skipConfirmation) {
```

Comment:
<!-- Questions comments seek clarification.  -->
**question (blocking):** ‏Is this the best user experience?

I think a user could be confused about how `--yes` works here, assuming it would update the git ignore file.

Maybe instead of prompting for this to happen, we show a command the user can copy-paste to do the same thing, making sure it changed based on current OS.

> You can add this with: `echo "my-env-file.env" >> ../path-to-git-root/.gitignore`

---

If we keep this flow, I share [Jeff's concerns](https://github.com/vercel/vercel/pull/12088#pullrequestreview-2290813854) about how this works in CI. We'd probably want to skip confirmation in CI.

Reply:
Note: If we do this, we'd also want to update `vercelignore` to make sure those secrets don't get uploaded. This won't matter in the basic case, but if the user updates that file after pulling, it would.